### DISCUSSION TITLE
Zero cost exception handling

### DISCUSSION THREAD

**Discussion by AliRezaBeigy** [dotnet/runtime#60237] [Ideas] [answered]
Hello everyone

recently I read some news about "zero-cost exception handling"
As I realized C++ and java has this feature, and python is working on it for Python 3.11
In some blog and Stackoverflow conversation, I see c# doesn't have this feature until now
I think it's a good idea to have a plan for this feature :D

**Comment by Clockwork-Muse** [comment, marked as answer]
C# already has near-zero costs for exceptions that are not thrown, which is what "zero cost" is referring to (not for exceptions that end up being thrown).  What makes you think C# doesn't have this feature?

**Reply by AliRezaBeigy** [reply]
You are right. First time I did not understand what it is. As you say, there is no zero-cost exception handling, it's near-zero cost exception handling, and it's referring to a non-exceptional case

**Reply by huoyaoyuan** [reply]
The cost of non-exceptional case of .NET is already very small. It could better if any measurements available.
